hubspot-style/src/main/java/com/hubspot/immutables/utils/WireSafeEnum.java

```diff
@@ -276,12 +277,38 @@ private static <T extends Enum<T>> void initializeCache(Class<T> enumType) {
       if (enumValue == deserializedValue) {
         jsonMap.put(jsonValue, wireSafeEnum);
       }
+
+      for (String alias : getJsonAliases(enumType, enumValue)) {
+        jsonMap.computeIfAbsent(
+          alias,
+          jsonAlias -> new WireSafeEnum<>(enumType, jsonAlias, enumValue)
+        );
+      }
     }
 
     ENUM_LOOKUP_CACHE.put(enumType, enumMap);
     JSON_LOOKUP_CACHE.put(enumType, jsonMap);
   }
 
+  private static <T extends Enum<T>> String[] getJsonAliases(
+    Class<T> enumType,
+    T enumConstant
+  ) {
+    JsonAlias annotation;
+
+    try {
+      annotation = enumType.getField(enumConstant.name()).getAnnotation(JsonAlias.class);
```

Comment:
could there be multiple `@JsonAlias` annotations on the same constant?

Reply:
It doesn't have `@Repeatable` on it, so I don't think so

Reply:
cool nice